### DISCUSSION TITLE
Unable to deploy milvus on Azure

### DISCUSSION THREAD

**Discussion by RGEsteves** [milvus-io/milvus#35885] [Q&A and General discussion]
Hi

I'm unable to deploy milvus on Azure. I tried to follow the documentation with Kubernetes and even tried to deploy with docker in a container instance and in a web app with docker compose config file.

With the kubernetes the etcd pod was never started. Can't understand why!
On the web app, with the default compose config file, it fails with etcd too (or at least seems to)!

I event try to read milvus code to get some light but without success.

Can someone help me pls ? I liked to use milvus locally but if I can't deploy on Azure is useless in my case.

Thanks in advance.

**Comment by xiaofan-luan** [comment]
try https://milvus.io/docs/azure.md see if it help.

1. you need several ec2 under kubernetesto deploy etcd. that is usually same as oss milvus.
2. you have modify the helm chart to use azure storage rather than S3.
3. we need deployed log about what happened to your cluster if it still failed
you always have the choice to use zilliz cloud https://zilliz.com/pricing
